components/toggle_modal_button/toggle_modal_button.tsx

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {ComponentType, MouseEvent, ReactNode} from 'react';
+import {useIntl} from 'react-intl';
+
+import {ModalData} from 'types/actions';
+
+type Props = {
+    ariaLabel?: string;
+    children: ReactNode;
+    modalId: string;
+    dialogType: ComponentType<any>;
+    dialogProps?: Record<string, any>;
```

Comment:
It'd be nice if we were able to get these to be fully typed so that the ToggleModalButton would ensure that the correct props were passed for each type of modal, but making a React component with a type parameter has never worked out for me. This component using `connect` also makes that difficult as well.

Regardless, this looks good to me!

Reply:
Exactly, `connect` makes even more hard for me to figure out the exact types